### PR TITLE
Tweaks Floorbot pathfinding to prevent dogpiles and help prevent duplicate tile placements.

### PR DESCRIPTION
**Pull request by ArcaneMusic**
## About The Pull Request

This pr adjusts floorbots behaviors in order to attempt to fix any of the following issues:
#54159, #54196, #37421.
More or less the issue appears to have been that floorbots would place floor platings on top of where another floorbot would already be repairing a tile, and since they don't check for each other's work, they were stacking floor turfs ontop of each other, creating either A: Infinite piles of floor tiles, or B: Multiple floor tiles stacked on top of each other. The former I think may have been fixed in the previous changes to how shuttlecode turfs were stacking on-top of each other, but the multiple stacks of  floor tiles overlaying was not resolved.

This adds a check to the beginning of floorbots repair proc which determines if another floorbot is there and repairing the turf already. If so, it begins to ignore the turf until it is reset next, which it tends to go over the course of a single job. This helps split up and divide multiple floorbots working on the same tile at a time, in order to prevent infinite floorstacks.

If 2 floorbots are on the same tile and conflicting, then a little whistle effect will play to indicate that they're moving to a new worksite.

## Why It's Good For The Game

This bug is the absolute bane of of the issue section and seems to come back in some fashion every few months. In lou of wanting to rework all of simplebots and in general just base mob pathfinding, this is a clean bandaid fix that will prevent them from creating little pocketdimensions where they can duplicate floors.

## Changelog
:cl:
fix: Floorbots had had a software update, preventing them from dogpiling on their target as easily as they did before.
soundadd: Floorbots will now play a small chime when stacked on top of each other to indicate that they're moving apart.
/:cl:

### CODE REVIEW

**Comment by ArcaneMusic** [comment]
Okay I must have fucked something on my master, hold on.

File: code/modules/mob/living/simple_animal/bot/floorbot.dm [hunk 1]
```diff
@@ -393,3 +396,16 @@
...
+
+/**
+  * Checks a given turf to see if another floorbot is there, working as well.
+  */
+/mob/living/simple_animal/bot/floorbot/proc/check_bot_working(work_turf)
+	var/turf/active_turf
+	if(isturf(work_turf))
+		active_turf = work_turf
+	if(active_turf)
+		for(var/mob/living/simple_animal/bot/floorbot/robot in active_turf.contents)
```
Comment (Cyberboss) [inline code/modules/mob/living/simple_animal/bot/floorbot.dm:408]:
```suggestion
/mob/living/simple_animal/bot/floorbot/proc/check_bot_working(turf/active_turf)
	if(isturf(active_turf))
		for(var/mob/living/simple_animal/bot/floorbot/robot in active_turf)
```

File: code/modules/mob/living/simple_animal/bot/floorbot.dm [hunk 2]
```diff
@@ -309,7 +308,11 @@
 	return result
 
 /mob/living/simple_animal/bot/floorbot/proc/repair(turf/target_turf)
-
+	if(check_bot_working(target_turf))
+		add_to_ignore(target_turf)
+		target = null
+		playsound(src.loc, 'sound/effects/whistlereset.ogg', 50, TRUE)
```
Comment (Cyberboss) [inline code/modules/mob/living/simple_animal/bot/floorbot.dm:314]:
```suggestion
		playsound(src, 'sound/effects/whistlereset.ogg', 50, TRUE)
```

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Review by Cyberboss — APPROVED** [review]

**Comment by TemporalOroboros** [comment]
This didn't actually close any of the linked issues.

**Comment by ArcaneMusic** [comment]
I'm aware of that, because I'm not fully sure that I actually solved those issues.
I'm going to give it a few days to see if the issue still pops up (Since it seems to re-open itself every few days), and if it's working smoothly I'll go through and close them.

**Pull request merged**